### PR TITLE
Add unit tests for Choropleth base component

### PR DESCRIPTION
**Pull request by VWSCoronaDashboard1**
## Summary

Since we are now able to run jest tests successfully, these are some first unit tests

## Motivation

Testing is good

## Detailed design

These tests make sure that the correct callbacks are invoked for the correct feature collections. They check if the onclick handler is called with the correct corresponding id and if the click or tooltip is triggered correctly based on the screen size.
It adds a small test helper called mockMatchMedia that can help developers mock a specific screen size.

Choropleth is a rather large component, so it needs quite a bit of scaffolding to setup.

## Drawbacks

n/a

## Alternatives

n/a

## Unresolved questions

n/a

### Governance

- [ ] Documentation is added
- [ ] Test cases are added or updated
- [ ] I've read the contributing document https://github.com/minvws/.github/blob/master/CONTRIBUTING.md
- [ ] I've read and understand the Code of Conduct https://github.com/minvws/.github/blob/master/CODE_OF_CONDUCT.md
- [ ] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License
      https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/LICENSE.txt
      and the contributor license agreement https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/CLA.md

### CODE REVIEW

File: src/components/chloropleth/MunicipalityChloropleth.tsx [hunk 1]
```diff
@@ -182,7 +182,7 @@ export function MunicipalityChloropleth<
         SafetyRegionProperties | GeoJsonProperties
       >,
       path: string,
-      index: number
+      _index: number
```
Comment (VWSCoronaDashboard8) [inline src/components/chloropleth/MunicipalityChloropleth.tsx:185]: The use of _ to mark unused arguments is good, but if it's the last argument I think you might as well remove it altogether in Javascript.

File: src/components/chloropleth/__tests__/Chloropleth.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+import { MunicipalityProperties } from '../shared';
+import { Feature, MultiPolygon } from 'geojson';
+import { countryGeo, municipalGeo } from '../topology';
+import { mockMatchMedia } from '~/utils/testhelpers/mockMatchMedia';
+
+describe('Component: Choropleth', () => {
+  const featureCollection = municipalGeo;
+  const overlays = countryGeo;
+  const hovers = municipalGeo;
+  const boundingbox = municipalGeo;
```
Comment (VWSCoronaDashboard8) [inline src/components/chloropleth/__tests__/Chloropleth.spec.tsx:13]: nit; boundingBox

File: src/components/chloropleth/__tests__/Chloropleth.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+    _path: string,
+    index: number
+  ) => {
+    return <path key={`overlay-${index}`} />;
+  };
+
+  const defaultHoverCallback = (
+    feature: Feature<MultiPolygon, MunicipalityProperties>,
+    _path: string,
+    _index: number
```
Comment (VWSCoronaDashboard8) [inline src/components/chloropleth/__tests__/Chloropleth.spec.tsx:35]: Both of these can be removed, but I don't mind if you leave them in.

File: src/components/chloropleth/__tests__/Chloropleth.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+  it('Should call the correct callbacks for the given feature collections', () => {
+    // Arrange
+    let featureCallbackWasCalled = false;
+    let overlayCallbackWasCalled = false;
+    let hoverCallbackWasCalled = false;
+
+    const featureCallback = (
+      feature: Feature<MultiPolygon, MunicipalityProperties>,
+      _path: string,
+      _index: number
```
Comment (VWSCoronaDashboard8) [inline src/components/chloropleth/__tests__/Chloropleth.spec.tsx:49]: Same as last comment

File: src/components/chloropleth/__tests__/Chloropleth.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+    _index: number
+  ) => {
+    return <path key={`hover-${feature.properties.gemcode}`} />;
+  };
+
+  it('Should call the correct callbacks for the given feature collections', () => {
+    // Arrange
+    let featureCallbackWasCalled = false;
+    let overlayCallbackWasCalled = false;
+    let hoverCallbackWasCalled = false;
```
Comment (VWSCoronaDashboard8) [inline src/components/chloropleth/__tests__/Chloropleth.spec.tsx:44]: I don't know if what type of convention you use in this repo, but I'd start with `is`. The past/present tense doesn't matter much for booleans I think.

For example `isCalledFeatureCallback`. It fits more with booleans like `isDefined` or `hasMultipleValues`

File: src/components/chloropleth/__tests__/Chloropleth.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+      index: number
+    ) => {
+      overlayCallbackWasCalled = true;
+      expect(overlays.features.indexOf(feature)).toBeGreaterThan(-1);
+      return <path key={`overlay-${index}`} />;
+    };
+
+    const hoverCallback = (
+      feature: Feature<MultiPolygon, MunicipalityProperties>,
+      _path: string,
```
Comment (VWSCoronaDashboard8) [inline src/components/chloropleth/__tests__/Chloropleth.spec.tsx:68]: Similar comment about args

**Review by VWSCoronaDashboard — APPROVED** [review]
Please add tests to verify the DOM output as well, but besides that... YEEEEES!

File: src/components/chloropleth/__tests__/Chloropleth.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+    const renderResult = render(<Chloropleth {...props} />);
+
+    let groups = null;
+    try {
+      groups = renderResult.getAllByText(
+        (_content: string, element: HTMLElement) => {
+          return element.nodeName === 'g';
+        }
+      );
+    } catch (e) {}
```
Comment (VWSCoronaDashboard8) [inline src/components/chloropleth/__tests__/Chloropleth.spec.tsx:99]: Don't you at least want do something like print the error? 

In my experience, failing silently is only useful if you act on it in the catch clause.

File: src/components/chloropleth/__tests__/Chloropleth.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+
+    // Assert
+    expect(featureCallbackWasCalled).toBeTruthy();
+    expect(overlayCallbackWasCalled).toBeTruthy();
+    expect(hoverCallbackWasCalled).toBeTruthy();
+    // Check if there are exactly 4 <g> elements. One container, and one for each feature collection
```
Comment (VWSCoronaDashboard8) [inline src/components/chloropleth/__tests__/Chloropleth.spec.tsx:105]: The part of checking exactly 4 is clear. What happened to the other checks? I don't think this comment adds value, especially if it claims to be doing something it's not.

File: src/components/chloropleth/__tests__/Chloropleth.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+      return (
+        <path
+          data-id={feature.properties.gemcode}
+          data-testid={`feature-${feature.properties.gemcode}`}
+          key={`feature-${feature.properties.gemcode}`}
+        />
+      );
+    };
+
+    let pathReceiveClick = false;
```
Comment (VWSCoronaDashboard8) [inline src/components/chloropleth/__tests__/Chloropleth.spec.tsx:129]: This name doesn't read like a boolean. Maybe `hasPathReceivedClick`?

File: src/components/chloropleth/__tests__/Chloropleth.spec.tsx [hunk 10]
```diff
@@ -0,0 +1,270 @@
...
```
Comment (VWSCoronaDashboard8) [inline src/components/chloropleth/__tests__/Chloropleth.spec.tsx:180]: Same comment for this one

File: src/components/chloropleth/__tests__/Chloropleth.spec.tsx [hunk 11]
```diff
@@ -0,0 +1,270 @@
...
+    const onPathClick = (id: string) => {
+      pathReceiveClick = true;
+      expect(id).toEqual(testMunicipalCode);
+    };
+
+    let tooltipRequested = false;
```
Comment (VWSCoronaDashboard8) [inline src/components/chloropleth/__tests__/Chloropleth.spec.tsx]:
```suggestion
    let isTooltipRequested = false;
```

File: src/components/chloropleth/__tests__/Chloropleth.spec.tsx [hunk 12]
```diff
@@ -0,0 +1,270 @@
...
+  it('Should call the tooltip content handler when paths are mouse-overed', () => {
+    // Arrange
+    const testMunicipalCode = 'GM0003';
+
+    mockMatchMedia(true);
+
+    const featureCallback = (
+      feature: Feature<MultiPolygon, MunicipalityProperties>,
+      _path: string,
+      _index: number
```
Comment (VWSCoronaDashboard8) [inline src/components/chloropleth/__tests__/Chloropleth.spec.tsx:226]: Args again. I personally don't see a point in writing them out if not used.

**Review by VWSCoronaDashboard8 — CHANGES_REQUESTED** [review]

File: src/components/chloropleth/__tests__/Chloropleth.spec.tsx (hunk 2 above)
Comment (VWSCoronaDashboard1) [reply src/components/chloropleth/__tests__/Chloropleth.spec.tsx]: I agree :)

File: src/components/chloropleth/__tests__/Chloropleth.spec.tsx (hunk 5 above)
Comment (VWSCoronaDashboard1) [reply src/components/chloropleth/__tests__/Chloropleth.spec.tsx]: These were removed entirely in favor of a more practical way of testing. I don't really care if these call backs were called, I just want to test the result. So, now I'm just testing for the presence of the elements in the DOM.

File: src/components/chloropleth/__tests__/Chloropleth.spec.tsx (hunk 8 above)
Comment (VWSCoronaDashboard1) [reply src/components/chloropleth/__tests__/Chloropleth.spec.tsx]: This has been updated with a different check

File: src/components/chloropleth/__tests__/Chloropleth.spec.tsx (hunk 11 above)
Comment (VWSCoronaDashboard1) [reply src/components/chloropleth/__tests__/Chloropleth.spec.tsx]: This was also removed in favor of a DOM test

**Pull request merged**